src/lib/mdns/Resolver.h

```diff
@@ -138,6 +141,66 @@ struct DiscoveredNodeData
         return mrpRetryIntervalActive != kUndefinedRetryInterval ? Optional<uint32_t>{ mrpRetryIntervalActive }
                                                                  : Optional<uint32_t>{};
     }
+
+    void LogDetail() const
+    {
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+        if (rotatingIdLen > 0)
+        {
+            char rotatingIdString[chip::Mdns::kMaxRotatingIdLen * 2 + 1] = "";
+            Encoding::BytesToUppercaseHexString(rotatingId, rotatingIdLen, rotatingIdString, sizeof(rotatingIdString));
+            ChipLogDetail(Discovery, "Rotating ID: %s", rotatingIdString);
+        }
+#endif // CHIP_ENABLE_ROTATING_DEVICE_ID
+        if (strlen(deviceName) != 0)
+        {
+            ChipLogDetail(Discovery, "Device Name: %s", deviceName);
+        }
+        if (vendorId > 0)
+        {
+            ChipLogDetail(Discovery, "Vendor ID: %u", vendorId);
+        }
+        if (productId > 0)
+        {
+            ChipLogDetail(Discovery, "Product ID: %u", productId);
+        }
+        if (deviceType > 0)
+        {
+            ChipLogDetail(Discovery, "Device Type: %u", deviceType);
+        }
+        if (longDiscriminator > 0)
+        {
+            ChipLogDetail(Discovery, "Long Discriminator: %u", longDiscriminator);
+        }
+        if (additionalPairing > 0)
+        {
+            ChipLogDetail(Discovery, "Additional Pairing: %u", additionalPairing);
+        }
+        if (strlen(pairingInstruction) != 0)
+        {
+            ChipLogDetail(Discovery, "Pairing Instruction: %s", pairingInstruction);
+        }
+        if (pairingHint > 0)
+        {
+            ChipLogDetail(Discovery, "Pairing Hint: 0x%x", pairingHint);
+        }
+        if (!IsHost(""))
+        {
+            ChipLogDetail(Discovery, "Hostname: %s", hostName);
+        }
+        for (int j = 0; j < numIPs; j++)
+        {
+            char buf[Inet::kMaxIPAddressStringLength];
+            char * ipAddressOut = ipAddress[j].ToString(buf, sizeof(buf));
+            ChipLogDetail(Discovery, "IP Address #%d: %s", j + 1, ipAddressOut);
+            (void) ipAddressOut;
```

Comment:
Does this allow the ToString call to be optimized out when detail logging is disabled?  If not, this should use the appropriate detail logging ifdefs instead.  Followup is fine for this, but please make sure it happens.